dashboard/modules/job/tests/test_job_manager.py

```diff
@@ -115,6 +116,53 @@ async def test_submit_no_ray_address(call_ray_start, tmp_path):  # noqa: F811
     )
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["ray start --head"],
+    indirect=True,
+)
+async def test_get_all_job_info(call_ray_start, tmp_path):  # noqa: F811
+    """Test that JobInfo is correctly populated in the GCS get_all_job_info API."""
+    address_info = ray.init(address=call_ray_start)
+    gcs_aio_client = GcsAioClient(
+        address=address_info["gcs_address"], nums_reconnect_retry=0
+    )
+    job_manager = JobManager(gcs_aio_client, tmp_path)
+
+    # Submit a job.
+    submission_id = await job_manager.submit_job(
+        entrypoint="python -c 'import ray; ray.init()'",
+    )
+
+    # Wait for the job to be finished.
+    await async_wait_for_condition_async_predicate(
+        check_job_succeeded, job_manager=job_manager, job_id=submission_id
+    )
+
+    found = False
+    for job_table_entry in (await gcs_aio_client.get_all_job_info()).job_info_list:
+        if job_table_entry.config.metadata.get(JOB_ID_METADATA_KEY) == submission_id:
```

Comment:
not really sure which is more clear, but you can also use the `for: else:` syntax to achieve this in python:

`for ...:
    if condition: break
else:
    raise RuntimeError("thing not found")
```

(the `else` block only executes if the loop exits normally)

Reply:
Interesting! For-else looks weird to my eye for now, but I may start using it in the future

Reply:
I used to think so too and at some point I started liking it...